### ISSUE TITLE
F# files show project-level warnings as errors on first line of file

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#5688]
In F# files, project-level warnings (e.g. missing files) are shown as errors on the first line of every file in the project. In addition, each problem is shown twice.

![image](https://user-images.githubusercontent.com/183285/44111828-814861a6-9fd1-11e8-99e3-6006af7b987a.png)

In the error pad they have very different locations, and they are warnings.

![image](https://user-images.githubusercontent.com/183285/44111942-c5cd8856-9fd1-11e8-93cd-9031ed7258dc.png)

**Comment by nosami** [comment]
I think this is a compiler issue. The error pad is showing project-wide errors, whilst the editor is showing document wide issues. We show whatever the compiler reports.

**Comment by mhutch** [comment]
Can we filter out the project-wide errors i.e. errors without a location?

**Comment by nosami** [comment]
I'm not sure. I think sometimes these can be genuine errors rather than warnings.

But maybe that doesn't matter. A build would show these up.

**Comment by nosami** [comment]
Or maybe we need to filter out error codes beginning with `MSB` ?

**Comment by mhutch** [comment]
Yeah, it's not really the appropriate place to show those kinds of errors. There's a problem with the project, not the code in the file.

Where would I find this to fix it?

**Comment by nosami** [comment]
It should probably be fixed in the compiler, but you can filter them out [here](https://github.com/mono/monodevelop/blob/8d12b016bae602593a1419f45cfcd3c23cb022a7/main/external/fsharpbinding/MonoDevelop.FSharpBinding/FSharpParser.fs#L40) I think
